invenio/modules/oauthclient/contrib/cern.py

```diff
@@ -0,0 +1,205 @@
+
+       CERN_APP_CREDENTIALS = dict(
+           consumer_key="<CLIENT ID>",
+           consumer_secret="<CLIENT SECRET>",
+       )
+
+4. Now go to ``CFG_SITE_SECURE_URL/oauth/login/cern/`` (e.g.
+   http://localhost:4000/oauth/login/cern/)
+
+5. Also, you should see CERN listed under Linked accounts:
+   http://localhost:4000//account/settings/linkedaccounts/
+
+By default the CERN module will try first look if a link already exists
+between a CERN account and a user. If no link is found, the user is asked
+to provide an email address to sign-up.
+
+In templates you can add a sign in/up link:
+
+.. code-block:: jinja
+
+    <a href="{{url_for("oauthclient.login", remote_app="cern")}}">
+      Sign in with CERN
+    </a>
+
+"""
+
+import copy
+import re
+import requests
+
+from flask import current_app
+
+from flask_login import current_user
+
+#: Tunable list of groups to be hidden.
+CFG_EXTERNAL_AUTH_HIDDEN_GROUPS = (
+    'All Exchange People',
+    'CERN Users',
+    'cern-computing-postmasters',
+    'cern-nice2000-postmasters',
+    'CMF FrontEnd Users',
+    'CMF_NSC_259_NSU',
+    'Domain Users',
+    'GP Apply Favorites Redirection',
+    'GP Apply NoAdmin',
+    'info-terminalservices',
+    'info-terminalservices-members',
+    'IT Web IT',
+    'NICE Deny Enforce Password-protected Screensaver',
+    'NICE Enforce Password-protected Screensaver',
+    'NICE LightWeight Authentication WS Users',
+    'NICE MyDocuments Redirection (New)',
+    'NICE Profile Redirection',
+    'NICE Terminal Services Users',
+    'NICE Users',
+    'NICE VPN Users',
+    )
+
+#: Tunable list of regexps of groups to be hidden.
+CFG_EXTERNAL_AUTH_HIDDEN_GROUPS_RE = (
+    re.compile(r'Users by Letter [A-Z]'),
+    re.compile(r'building-[\d]+'),
+    re.compile(r'Users by Home CERNHOME[A-Z]'),
+    )
+
+REMOTE_APP = dict(
+    title="CERN",
+    description="Connecting to CERN Organization.",
+    icon="",
+    authorized_handler="invenio.modules.oauthclient.handlers"
+                       ":authorized_signup_handler",
+    disconnect_handler="invenio.modules.oauthclient.handlers"
+                       ":disconnect_handler",
+    signup_handler=dict(
+        info="invenio.modules.oauthclient.contrib.cern:account_info",
+        setup="invenio.modules.oauthclient.contrib.cern:account_setup",
+        view="invenio.modules.oauthclient.handlers:signup_handler",
+    ),
+    params=dict(
+        request_token_params={
+            "resource": "datadev00.cern.ch",
+            "scope": "Name Email Bio Groups",
+        },
+        base_url="https://oauth.web.cern.ch/",
+        request_token_url=None,
+        access_token_url="https://oauth.web.cern.ch/OAuth/Token",
+        access_token_method="POST",
+        authorize_url="https://oauth.web.cern.ch/OAuth/Authorize",
+        app_key="CERN_APP_CREDENTIALS",
+        content_type="application/json",
+    )
+)
+"""CERN Remote Application."""
+
+REMOTE_SANDBOX_APP = copy.deepcopy(REMOTE_APP)
```

Comment:
I think that one is not working (at least during my test). The provided URLs at 149++ are for a oauth test client and not for a oauth test provider. I suggest to remove the sandbox app from this file.